nni/algorithms/compression/v2/pytorch/base/compressor.py

```diff
@@ -0,0 +1,285 @@
+
+        Parameters
+        ----------
+        layer
+            The layer that need to check if has compression configuration.
+
+        Returns
+        -------
+        Optional[Dict]
+            The retrieved configuration for this layer, if None, this layer should not be compressed.
+        """
+        ret = None
+        for config in self.config_list:
+            config = config.copy()
+            # expand config if key `default` is in config['op_types']
+            if 'op_types' in config and 'default' in config['op_types']:
+                expanded_op_types = []
+                for op_type in config['op_types']:
+                    if op_type == 'default':
+                        expanded_op_types.extend(weighted_modules)
+                    else:
+                        expanded_op_types.append(op_type)
+                config['op_types'] = expanded_op_types
+
+            # check if condition is satisified
+            if 'op_types' in config and layer.type not in config['op_types']:
+                continue
+            if 'op_names' in config and layer.name not in config['op_names']:
+                continue
+
+            ret = config
+        if ret is None or 'exclude' in ret:
+            return None
+        return ret
+
+    def get_modules_wrapper(self) -> OrderedDict[str, Module]:
+        """
+        Returns
+        -------
+        OrderedDict[str, Module]
+            An ordered dict, key is the name of the module, value is the wrapper of the module.
+        """
+        return self.modules_wrapper
+
+    def _wrap_model(self):
+        """
+        Wrap all modules that needed to be compressed.
+        """
+        for _, wrapper in reversed(self.get_modules_wrapper().items()):
+            _setattr(self.bound_model, wrapper.name, wrapper)
+        self.is_wrapped = True
+
+    def _unwrap_model(self):
+        """
+        Unwrap all modules that needed to be compressed.
+        """
+        if self.is_wrapped:
+            for _, wrapper in self.get_modules_wrapper().items():
+                _setattr(self.bound_model, wrapper.name, wrapper.module)
+            self.is_wrapped = False
+
+    def set_wrappers_attribute(self, name: str, value: Any):
+        """
+        To register attributes used in wrapped module's forward method.
+        If the type of the value is Torch.tensor, then this value is registered as a buffer in wrapper,
+        which will be saved by model.state_dict. Otherwise, this value is just a regular variable in wrapper.
+
+        Parameters
+        ----------
+        name
+            Name of the variable.
+        value
+            Value of the variable.
+        """
+        for wrapper in self.get_modules_wrapper():
+            if isinstance(value, torch.Tensor):
+                wrapper.register_buffer(name, value.clone())
+            else:
+                setattr(wrapper, name, value)
+
+    def generate_graph(self, dummy_input: Tensor) -> TorchModuleGraph:
```

Comment:
Is it possible that the dummy_input is not a `Tensor`? Like we discussed this afternoon, the input to transformer encoders is often a two-element tuple.

Reply:
yes, in fact, the input type is determined by the model, change it to Any.